addons/mrp/tests/test_order.py

```diff
@@ -5427,6 +5427,22 @@ def test_final_product_as_component(self):
         mo.button_mark_done()
         self.assertEqual(self.env['stock.quant']._get_available_quantity(self.productA, self.stock_location, lot_id=serial_number), 1)
 
+    def test_copy_operation_excludes_variant(self):
+        """ Test that copying an operation does not copy the variant of the product in the operation. """
+        bom = self.env['mrp.bom'].create({
+            'product_tmpl_id': self.product_7_template.id,
+            'operation_ids': [Command.create({
+                'name': 'Weld Machine',
+                'workcenter_id': self.workcenter_1.id,
+                'bom_product_template_attribute_value_ids': [Command.link(self.product_7_attr1_v1.id)],
+            })]
+        })
+        operation = bom.operation_ids[0]
```

Comment:
```suggestion
        operation = bom.operation_ids
```
`bom.operation_ids` always contains exactly one operation.